java/src/main/java/io/cucumber/gherkin/GherkinLine.java

```diff
@@ -65,20 +63,26 @@ public String getRestTrimmed(int length) {
 
     @Override
     public List<GherkinLineSpan> getTags() {
-
-        String uncommentedLine = trimmedLineText.split("\\s" + COMMENT_PREFIX, 2)[0];
-        List<GherkinLineSpan> tags = new ArrayList<>();
+        // in most cases, the line contains no tag, so the code is optimized for this situation
+        if (emptyTrimmedLineText) {
```

Comment:
Can also be applied to `getTableCells`.

Reply:
No, because `getTableCells` is called by `TokenMatcher.match_TableRow()` and only if there is a table separator. Thus the line is never empty.